tmt/steps/report/junit.py

```diff
@@ -47,6 +47,39 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
             f"Malformed duration '{duration}' ({error}).")
 
 
+def make_junit_xml(report: "tmt.steps.report.ReportPlugin") -> "junit_xml.TestSuite":
```

Comment:
Please, try the following patch:

```patch
diff --git a/tmt/steps/report/junit.py b/tmt/steps/report/junit.py
index 4e89efa..63e7fed 100644
--- a/tmt/steps/report/junit.py
+++ b/tmt/steps/report/junit.py
@@ -1,6 +1,6 @@
 import os
 import types
-from typing import Any, List, Optional, overload
+from typing import Any, List, Optional, cast, overload
 
 import click
 
@@ -12,6 +12,11 @@ DEFAULT_NAME = "junit.xml"
 
 junit_xml: Optional[types.ModuleType] = None
 
+# Thanks to import burried in a function, we don't get the regular missing annotations
+# but "name ... is not defined". Define a special type to follow the test suite instance
+# around, and if junit_xml ever gets annotations, we can replace it with provided type.
+JunitTestSuite = Any
+
 
 def import_junit_xml() -> None:
     """
@@ -47,7 +52,7 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
             f"Malformed duration '{duration}' ({error}).")
 
 
-def make_junit_xml(report: "tmt.steps.report.ReportPlugin") -> "junit_xml.TestSuite":
+def make_junit_xml(report: "tmt.steps.report.ReportPlugin") -> JunitTestSuite:
     """ Create junit xml object """
     import_junit_xml()
     assert junit_xml
@@ -77,7 +82,7 @@ def make_junit_xml(report: "tmt.steps.report.ReportPlugin") -> "junit_xml.TestSu
         suite.test_cases.append(case)
 
     assert report.workdir is not None
-    return suite
+    return cast(JunitTestSuite, suite)
 
 
 class ReportJUnit(tmt.steps.report.ReportPlugin):
```

This should take resolve the "optionality" of the `junit_xml` import.